lib/dl_api_commons/dl_api_commons/httpx/client.py

```diff
@@ -0,0 +1,174 @@
+import abc
+import contextlib
+import logging
+import ssl
+from types import TracebackType
+from typing import (
+    Any,
+    AsyncGenerator,
+    Awaitable,
+    Callable,
+    Iterator,
+)
+
+import attrs
+import httpx
+from typing_extensions import Self
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseRetrier(abc.ABC):
+    @abc.abstractmethod
+    def retry_request(
+        self, req_func: Callable[..., httpx.Response], request: httpx.Request, *args: Any, **kwargs: Any
```

Comment:
let's define type `Callable[..., httpx.Response]` and `Callable[..., Awaitable[httpx.Response]]`